synapse/federation/transport/server.py

```diff
@@ -1466,11 +1495,13 @@ async def on_GET(self, origin, content, query, room_id):
     FederationQueryServlet,
     FederationMakeJoinServlet,
     FederationMakeLeaveServlet,
+    FederationMakeKnockServlet,
```

Comment:
Probably not worth fixing, but these should be guarded by the config flag?

Reply:
They probably should indeed (in case it tries to use other disabled knocking code).

More importantly, we should probably do the same for the client endpoints, especially if clients try to gauge whether knocking is enabled via looking for `M_UNRECOGNIZED_REQUEST` errors.